samply/src/linux_shared/mod.rs

```diff
@@ -399,7 +417,54 @@ where
         }
     }
 
+    /// This is a terrible hack to get binary correlation working with apps on Wine.
+    ///
+    /// Unlike ELF, PE has the notion of "file alignment" that is different from page alignment.
+    /// Hence, even if the virtual address is page aligned, its on-disk offset may not be. This
+    /// leads to obvious trouble with using mmap, since mmap requires the file offset to be page
+    /// aligned. Wine's workaround is straightforward: for misaligned sections, Wine will simply
+    /// copy the image from disk instead of mmaping them.
+    ///
+    /// When this misalignment happens, most of the sections in the memory will not be a file
+    /// mapping. However, the PE header is always mapped, and it resides at the beginning of the
+    /// file, which means it's also always *aligned*. Hence, if we find a mapping that seems to
+    /// pointing to a PE file, and has a file offset of 0, we'll add it to the list of "suspected
+    /// PE images". When we see a later mapping that belongs to one of the suspected PE ranges,
+    /// we'll match the mapping with the file, which allows binary correlation and unwinding to
+    /// work.
+    fn check_for_pe_mapping(&mut self, path_slice: &[u8], base_avma: u64) {
```

Comment:
Let's call the `base_avma` parameter `mapping_start_avma` instead - at this point we don't know yet whether it's actually the base AVMA of an image.